libevm/VM.cpp

```diff
@@ -1312,20 +1397,22 @@ void VM::interpretCases()
             updateMem(memNeed(m_SP[0], m_SP[2]));
             updateIOGas();
 
-            copyDataToMemory(&m_ext->code, m_SP);
+            copyDataToMemory({m_pCode, m_codeSize}, m_SP);
         }
         NEXT
 
         CASE(EXTCODECOPY)
         {
             ON_OP();
-            m_runGas = toInt63(m_schedule->extcodecopyGas);
+            m_runGas = m_rev >= EVM_TANGERINE_WHISTLE ? 700 : 20;
             m_copyMemSize = toInt63(m_SP[3]);
             updateMem(memNeed(m_SP[1], m_SP[3]));
             updateIOGas();
 
-            Address a = asAddress(m_SP[0]);
-            copyDataToMemory(&m_ext->codeAt(a), m_SP + 1);
+            evm_address address = toEvmC(asAddress(m_SP[0]));
+            uint8_t const* pCode = nullptr;
+            size_t codeSize = m_context->fn_table->get_code(&pCode, m_context, &address);
```

Comment:
Does this makes sure that `pCode` is not dereferenced if `codeSize` is 0?

Reply:
Here the `pCode` is just an output argument. But in the next line, copyDataToMemory will not do anything with it if the size is 0.

Reply:
This comment was meant for the `copyDataToMemory` line.

Reply:
Is this addressed in your opinion @axic ?

Reply:
Yes